### DISCUSSION TITLE
Send Response as Part of Delimited Text

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5032] [Q&A]
I have a HTTP sender destination that runs an API call that returns a CCD document. I am trying to send that CCD response over to a 2nd channel as part of a delimited text string along with another variable.

The issue I am having is that I need the CCD document to be all on one long continuous line. What I have tried is below in my response for the API call destination is below. For some reason however this is not putting the CCD completely on one line and is causing some weird spacing. Does anyone know what could be wrong here?

var outbound = response.getMessage()
channelMap.put('outbound', outbound.replaceAll("\n", ''));

**Comment by rjkroll** [comment]
Are you sure there are no Carriage Returns you need to check for?   Copy and past the CCD response into NotePad++ and turn on 'Show all characters' to see if you are missing anything to Replace

**Reply by JJWest12** [reply]
It looks to all be CRLF. Is it possible that the replaceAll is not working as intended because the type of the CCD I am getting back is an object rather than XML? It looks like XML would however when I go a typeof on the API response it claims it to be an object rather than xml

**Reply by tonygermano** [reply]
`\n` is a LF only. The CR is represented by `\r`.

**Reply by JJWest12** [reply]
Correct you are! Adding in the replace \r got my CCD to be all on 1 line! The last issue I am not running in to is that when I go to send the delimited text (variable, CCD) down to my 2nd channel it is reading it as 1 row but splitting the CCD in to multiple columns still despite column in on 1 line

EDIT: After looking at some things it looks like the CCD is breaking in to multiple columns because there are commas in the human readable sections of the CCD. Looking in to how to fix that now

**Reply by pacmano1** [reply]
Why are you doing this at all?   If you need to send an XML document as a single string, encode it, don't search and replace.

You can also encapsulate it in JSON and route to the next channel

**Reply by JJWest12** [reply]
I am doing it this way because I need to send the XML response to a 2nd channel along with a variable that is declared in the source transformer. I have it set up now to send to the 2nd channel the way I want as a delimited text string formatted like 'variable^^^XML'.

I now am currently at an issue where for some reason when I take in that delimited text in my 2nd channel and declare the CCD from the delimited text like such:

var ccd = new XML(msg['row']['column2'].toString())

When I try doing something like 

var total= ccd['component'].length();
channelMap.put('total', total)

That it is no recognizing the XML and 'total' is being blank. I think it may have something to do with how my data types are set up?

**Reply by pacmano1** [reply]
You have Message Metadata for that -  you can send your var with the message to the next channel:

<img width="1554" alt="image" src="https://user-images.githubusercontent.com/44065187/155046046-18b593a1-3bad-47f2-9806-7d1e9d0c77cd.png">

What you put in there will be available in the next channel.  You can just pass the entire XML as is.  You would put your variable in that map in the screenshot and retrieve it on the other channel
